source/include/TypeAndDataManager.h

```diff
@@ -951,8 +951,12 @@ class RefNumValType : public WrappedType
         *(RefNumValType**)pData = this;
         return kNIError_Success;
     }
-    virtual NIError ClearData(void* pData) { return kNIError_Success; }
-    virtual void*   Begin(PointerAccessEnum mode)       { return (void*)&_refnum; }
+    virtual NIError CopyData(const void* pData, void* pDataCopy)  {
+        *(RefNumValType**)pDataCopy = *(RefNumValType**)pData;
```

Comment:
Just a question, here I assume we copy the pointer to the class, I wonder if we need to create another instance and then copy the data from one to the other, actually creating a duplicate. Does it make sense?

Reply:
I noticed this when I added CopyData too.  I think this probably warrants more investigation.
This works currently because of the way we generate Vireo from DFIR, where every call to Obtain_Queue has its own unique storage, but this implementation could have an aliasing problem.

Reply:
If possible, I would recommend adding potential tests that may fail due to aliasing issues (and keep them disabled) -- while this is fresh in the mind.

Reply:
I verified we can get weird aliasing behavior.  I'll work on making a test out of it.  I may wait to submit it until I have a fix as well.

Reply:
Refactored refnum storage to avoid aliasing behavior so that queue refnums work inside arrays and clusters.